File: lib/rgfa/line/segment/gfa1.rb
# A segment line of a RGFA file
class RGFA::Line::Segment::GFA1 < RGFA::Line::Segment

  RECORD_TYPE = :S
  POSFIELDS = [:name, :sequence]
  PREDEFINED_TAGS = [:LN, :RC, :FC, :KC, :SH, :UR]
  DATATYPE = {
    :name => :segment_name_gfa1,
    :sequence => :sequence_gfa1,
    :LN => :i,
    :RC => :i,
    :FC => :i,
    :KC => :i,
    :SH => :H,
    :UR => :Z,
  }
  FIELD_ALIAS = { :sid => :name }
  REFERENCE_FIELDS = []
  REFERENCE_RELATED_FIELDS = []
  DEPENDENT_LINES = [:dovetails_L, :dovetails_R,
                     :edges_to_contained, :edges_to_containers]
  gfa2_compatibility = [:gaps_L, :gaps_R, :fragments, :internals, :sets]
  OTHER_REFERENCES = [:paths] + gfa2_compatibility

  define_field_methods

  alias_method :to_sym, :name

end

require_relative "gfa1_to_gfa2"
require_relative "length_gfa1"
require_relative "coverage"
require_relative "references"
require_relative "writer_wo_sequence"

class RGFA::Line::Segment::GFA1
  include RGFA::Line::Segment::GFA1ToGFA2
  include RGFA::Line::Segment::LengthGFA1
  include RGFA::Line::Segment::Coverage
  include RGFA::Line::Segment::References
  include RGFA::Line::Segment::WriterWoSequence
end

File: gfapy/line/segment/gfa1.py
from .gfa1_to_gfa2 import GFA1ToGFA2
from .length_gfa1 import LengthGFA1
from .coverage import Coverage
from .references import References
from .writer_wo_sequence import WriterWoSequence
from . import Segment

class GFA1(WriterWoSequence, References, Coverage,
           LengthGFA1, GFA1ToGFA2, Segment):
  """
  A segment line of a GFA file
  """

  VERSION = "gfa1"
  RECORD_TYPE = "S"
  POSFIELDS = ["name", "sequence"]
  PREDEFINED_TAGS = ["LN", "RC", "FC", "KC", "SH", "UR"]
  DATATYPE = {
    "name" : "segment_name_gfa1",
    "sequence" : "sequence_gfa1",
    "LN" : "i",
    "RC" : "i",
    "FC" : "i",
    "KC" : "i",
    "SH" : "H",
    "UR" : "Z",
  }
  NAME_FIELD = "name"
  FIELD_ALIAS = { "sid" : "name" }
  DEPENDENT_LINES = ["dovetails_L", "dovetails_R",
                     "edges_to_contained", "edges_to_containers", "paths"]
  gfa2_compatibility = ["gaps_L", "gaps_R", "fragments", "internals", "sets"]
  OTHER_REFERENCES = gfa2_compatibility

GFA1._apply_definitions()
